labs/b8.md

```diff
@@ -0,0 +1,84 @@
+
+![alt](https://cdn-images-1.medium.com/max/1280/1*313akmm3vKOhK0bVNMxtSQ.png =400x250)
+
+To try it out:
+`gpg --sign [FILE]` to sign `[FILE]` with your private key. `gpg --verify [FILE].gpg` to verify that the file was signed by one of public keys on your keyring. 
+
+#### Lab Activity 
+*`wget` might be useful for this lab's activities?*
+1. Decrypt `file1.txt.gpg` with the password `ocfdecal` (*for real-life purposes, never store passwords in plaintext*). What are the decrypted contents of `file1.txt.gpg`?
+2. How does one import a key?
+3. How does one export a key to a file? (Hint: use `--armor`)
+4. How does one see all of the keys on his/her keyring?
+5. Use this private key `privkey` to decrypt `file2.txt.gpg` (*for real-life purposes, please actually keep your private keys secret*). What are the decrypted contents of `file2.txt.gpg`?
+
+## Hashing (Checksums)
+Hash functions deterministically map arbitrary-length data to a fixed-length hash (in other words, a fixed number of bytes). As a result, the latter can serve as a summary of the former if the former exceeds the latter in terms of byte length. For instance, if we download a 1GB file and want to verify its integrity, instead of re-downloading the entire file again, we can simply compute a 20B hash of the file on our end and compare it to the 20B hash of source.
+
+To try it out:
+1. `sha1sum [FILE]` to get the SHA1 hash of `[FILE]`.
+2. `md5sum [FILE]` to get the MD5 hash of `[FILE]`.
+
+There are [many hash functions](https://en.wikipedia.org/wiki/List_of_hash_functions), some of which are cryptographic hash functions and others not. Crytographic hash functions primarily differ from non-cryptographic hash functions in that they are difficult to forge. For instance, if someone has a hash on hand, it is difficult for them to create a pre-hash file that maps to the same hash. If you are interested in all of the properties of a cryptographic hash function,  [read here](https://en.wikipedia.org/wiki/Cryptographic_hash_function).
+
+### Lab Activity
+1. What is the MD5 hash of `file3.txt`?
+2. What is the SHA1 hash of the MD5 hash of `file3.txt`? In other words, what is `SHA1(MD5(file3.txt))`?
+
+## File Security
+The UNIX permission model has 3 components: permissions given to the file's (1) owning user, (2) owning group and (3) others/everyone else. Permissions themselves have 3 subcomponents: (1) read, (2) write and (3) execute, which are self-explanatory in respectively enforcing the ability to read, write or execute a file.
+
+![alt](http://www.comentum.com/images/permissions.jpg)
+
+To try it out:
+1. `ls -l` to view the permissions of all files in the current directory.
```

Comment:
Maybe say which column the permissions are in? It should be pretty clear what they look like given the image above, but I think it would be good to say which column shows the permissions